src/extension/ChronoController.hpp

```diff
@@ -67,62 +69,66 @@ namespace extension {
             // When we shutdown we notify so we quit now
             on<Shutdown>().then("Shutdown Chrono Controller", [this] {
                 const std::lock_guard<std::mutex> lock(mutex);
+                running = false;
                 wait.notify_all();
             });
 
             on<Always, Priority::REALTIME>().then("Chrono Controller", [this] {
                 // Acquire the mutex lock so we can wait on it
                 std::unique_lock<std::mutex> lock(mutex);
 
-                // If we have tasks to do
-                if (!tasks.empty()) {
+                if (running) {
```

Comment:
Is this needed? Won't the always itself just stop running if running were false?
To reduce the diff this could just be
```cpp
if(running) {
    return;
}
```

Reply:
I think I saw what I thought was the chrono controller continuing to run after shutdown, so I added this to force the point, but now that I think about it it may have been to do with the prolific always spam we saw before